### ISSUE TITLE
[loss] add reduction option to cross entropy loss module

### ISSUE THREAD

**Issue by pilkicTT** [tenstorrent/tt-forge-fe#792]
Add an option to `CrossEntropyLoss` module so that user can configure reduction (disabling it, using mean reduction or sum reduction).

Torch reference: https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html

**Comment by nvukobratTT** [comment]
Not yet, we're moving it to the backlog. The plan is to address these kinds of issues in the following weeks during the new training push.

**Comment by vs4848** [comment]
Hi, i'd like to work on this. I'm planning to add a reduction parameter to CrossEntropyLoss so it matches the other losses in the file, right now it just hardcodes mean reduction. i'll add reduction="mean" to __init__, swap the hardcoded ReduceAvg for reduce_loss(self.reduction, ...), and update the test to parametrize over mean and sum like test_l1_loss does. should be backward compatible since mean is the default.

Could I get assigned? i can start the PR once that's good.
